src/Controller/AppController.php

```diff
@@ -39,4 +40,17 @@ public function initialize()
         parent::initialize();
         $this->loadComponent('Flash');
     }
+
+    /**
+     * Before render callback.
+     *
+     * @param \Cake\Event\Event $event The beforeRender event.
+     * @return void
+     */
+    public function beforeRender(Event $event)
+    {
+        if ($this->viewClass === 'JsonView') {
+            $this->set('_serialize', true);
+        }
```

Comment:
> What about xmlview?

The check should be updated when same feature is implemented for XmlView too :)

> Also, what if the fully-qualified namespace is used?

RequestHandler sets the viewClass to `JsonView` so I added check for that. If someone is manually setting the fully namespaced class name or a custom class they should update this check accordingly. In general the app skeleton is meant to be customized as per ones needs.